lib/filesystem_utils.pm

```diff
@@ -509,4 +510,33 @@ sub validate_lsblk {
     return $errors;
 }
 
+=head2
+
+Generate xfstests subtests list
+This function translate test range to single tests, with xfstests test name format.
+Input an parameter with list of subtest name, Return a hash of test list.
+e.g. generate "xfs/001-003,xfs/005" into "{xfs/001 => 1, xfs/002 => 1, xfs/003 => 1, xfs/005 => 1}"
+
+=cut
+
+sub generate_xfstests_list {
+    my $raw_list = shift;
+    my @split_list = split(/,/, $raw_list);
+    my @test_list;
+    foreach my $test_item (@split_list) {
+        $test_item =~ m"\w+/";
+        my ($test_category, $test_num) = ($&, $');
+        if ($test_num =~ /^(\d{3})-(\d{3})$/) {
+            push(@test_list, map { $_ = "$test_category$_" } ($1 .. $2));
+        }
+        elsif ($test_num =~ /^\d{3}$/) {
+            push(@test_list, "$test_category$test_num");
+        }
+        else {
+            die "Invalid test list: $test_item";
```

Comment:
if..elsif are checking for `$test_num` but default condition here dies for invalid $item. i think this is inconsistent. Can we improve the die message?